### ISSUE TITLE
DMR does not transmit or recieve audio when usb audio routing is active.

### ISSUE THREAD

**Issue by grandmagoblin** [ADeadTrousers/android_device_Unihertz_Atom_LXL#16]
DMR app opens and is stable. Messaging works. When trying to speak on intercom, it looks like it is trying to transmit; mic button glows red when pressing and speaking. It also looks like it attempts to receive incoming audio; mic button will glow green when it detects that there is audio being transmitted on that channel. 

Tested with:
 - Phone A - Atom XL - Factory Stock
 - Phone B - Atom XL - Lineage OS 18.1

**Comment by clewisit** [comment]
Out of curiosity, have you tried using a headset plugged into the 3.5mm jack?  Have you tried a Bluetooth headset for receiving only (Bluetooth mic doesn't work with Intercom)?  Are you testing with adeadtrousers release, or one you compiled yourself?

**Comment by grandmagoblin** [comment]
I am testing with both releases - one I compiled using adeadtrousers' tree and guide and one compiled by adeadtrousers. I will try your recommendation and see what happens. I have soft bricked one of my Atom XL's in the process of testing VoLTE fixes, so it may be a moment before I can get to this.

**Comment by clewisit** [comment]
I'm going to be honest with you, I don't understand what the issue is.  On my phone, I can load the released ROM, go through setup, enable developer settings, and set the "disable USB audio routing" option, then start intercom and I get received audio on the phone speaker.  

So, what's different between my phone and yours?  I have the Atom XL TEE (non -European version).  Fresh install, restored stock firmware, followed the instructions to load the latest twrp and latest released ROM from May.  No other changes, nothing else loaded.

**Comment by ADeadTrousers** [comment]
You are right. "disable USB audio routing" still doeas the trick. Now I feel like an idiot. 
In the old version (Android 10) the app sounded a beep after "initializing" the channel. Now this seems to be gone and therefore I assumed the sound is gone all together. I never actually tested my two phones side by side until today.

I will update my thread over at xda accordingly.

In the meantime I want to keep this Issue to maybe dig deeper into the reason why the option is needed.

**Comment by clewisit** [comment]
That's funny, but it is completely understandable since that's how you knew it was working before.  

Regarding the reason why we have to do this, I can only assume that there must be a way to prevent a specific device from being recognized as an audio output.  Similar to the way the rtl-sdr driver has to be blacklisted so that SDR programs can use that device on Linux.

**Comment by grandmagoblin** [comment]
Wanted to stop in and provide an update, I am still looking into this. My other Atom XL is still a little borked. Thanks @clewisit for clearing the issue up. I will see if there is a way around the USB routing audio option.
